### ISSUE TITLE
Pylint does not properly process a ctypes variable's value field.

### ISSUE THREAD

**Issue by sidneycadot** [PyCQA/pylint#4896]
### Bug description

```python
"""Demonstrate pylint issue."""

import ctypes

x = ctypes.c_float(1.0)

# The line below produces a strange PyLint warning:
#
#   E1101: Class 'value' has no 'is_integer' member (no-member)
#
# At runtime at least, x.value yields a 'float' value.
# It seems pylint gets tripped up by some ctypes construct.

print(x.value.is_integer())
```

### Command used

```shell
pylint x.py
```

### Pylint output

```shell
x.py:14:6: E1101: Class 'value' has no 'is_integer' member (no-member)
```

### Expected behavior

I'd expect no warning.

### Pylint version

```shell
pylint 2.9.6
astroid 2.6.6
Python 3.9.6 (default, Aug 16 2021, 20:37:45) 
[GCC 8.3.0]
```

### OS / Environment

Debian
